### PR TITLE
CIVDT-738: Remove unused controller parameters

### PR DESCRIPTION
**Pull request by jacemale-hmrc**

### CODE REVIEW

File: test/controllers/AddressLookupControllerSpec.scala [hunk 1]
```diff
@@ -46,7 +46,6 @@ class AddressLookupControllerSpec extends ControllerSpecBase {
         mockAddressLookupService,
         errorHandler,
         messagesControllerComponents,
-        appConfig,
```
Comment (tvaidotas) [inline test/controllers/AddressLookupControllerSpec.scala:49]: auto code formatting updates indentations

File: test/controllers/UploadAuthorityControllerSpec.scala [hunk 2]
```diff
@@ -335,25 +335,25 @@ class UploadAuthorityControllerSpec extends ControllerSpecBase {
     "checkMode is false" should {
       "return link to Rep Duty Dan" in new Test {
         override val userAnswers: Option[UserAnswers] = Some(UserAnswers("credId").set(SplitPaymentPage, true).success.value)
-        val result: Call = controller.backLink(Duty, dan, Both, splitPayment = true)(dataRequest)
+        val result: Call = controller.backLink(Duty, Both, splitPayment = true)(dataRequest)
```
Comment (tvaidotas) [inline test/controllers/UploadAuthorityControllerSpec.scala:338]: auto code formatting

File: test/controllers/reasons/UnderpaymentReasonSummaryControllerSpec.scala [hunk 3]
```diff
@@ -52,7 +52,7 @@ class UnderpaymentReasonSummaryControllerSpec extends ControllerSpecBase {
     private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
 
     lazy val controller = new UnderpaymentReasonSummaryController(authenticatedAction, dataRetrievalAction, dataRequiredAction,
-      messagesControllerComponents, view, formProvider, ec)
+      messagesControllerComponents, view, formProvider)
```
Comment (tvaidotas) [inline test/controllers/reasons/UnderpaymentReasonSummaryControllerSpec.scala:55]: theres an import in this file that pulls in like 5 things

File: test/controllers/underpayments/UnderpaymentStartControllerSpec.scala [hunk 4]
```diff
@@ -49,7 +49,7 @@ class UnderpaymentStartControllerSpec extends ControllerSpecBase with ReusableVa
     )
 
     lazy val controller = new UnderpaymentStartController(authenticatedAction, dataRetrievalAction,
-      messagesControllerComponents, dataRequiredAction, view, ec)
+      messagesControllerComponents, dataRequiredAction, view)
```
Comment (tvaidotas) [inline test/controllers/underpayments/UnderpaymentStartControllerSpec.scala:52]: auto code formatting

**Review by tvaidotas — CHANGES_REQUESTED** [review]
some additional cleaning up could be done

**Review by tvaidotas — APPROVED** [review]

**Review by jakebooker56 — APPROVED** [review]

**Pull request merged**